pysat/tests/test_utils_io.py

```diff
@@ -829,3 +1207,405 @@ def test_pysat_meta_to_xarray_attr(self, export_nan):
+
+        # Apply default translation
+        self.out = io.apply_table_translation_to_file(self.test_inst,
+                                                      self.meta_dict)
+
+        # Shift values of _FillValue but not FillVal
+        for key in self.out.keys():
+            if '_FillValue' in self.out[key].keys():
+                self.out[key]['_FillValue'] += 1
+
+        # Get default inverse translation
+        from_trans = io.default_from_netcdf_translation_table(
+            self.test_inst.meta)
+
+        # Apply inverse
+        with caplog.at_level(logging.WARNING, logger='pysat'):
+            io.apply_table_translation_from_file(from_trans, self.out)
+
+        self.out = caplog.text
+        assert self.out.find('Inconsistent values between file and translated')
+
+        return
+
+    def test_from_file_table_translation_missing(self, caplog):
+        """Test `apply_table_translation_from_file` label not found message."""
+
+        # Apply default translation
+        self.out = io.apply_table_translation_to_file(self.test_inst,
+                                                      self.meta_dict)
+
+        # Get default inverse translation
+        from_trans = io.default_from_netcdf_translation_table(
+            self.test_inst.meta)
+
+        # Add a new label that is not present
+        from_trans['missing'] = ['not_found']
+
+        # Apply inverse
+        with caplog.at_level(logging.DEBUG, logger='pysat'):
+            io.apply_table_translation_from_file(from_trans, self.out)
+
+        self.out = caplog.text
+
+        for key in self.meta_dict.keys():
+            estr = ''.join(['Translation label "missing" not found for ',
+                            'variable "', key, '".'])
+            assert self.out.find(estr) >= 0
+
+        return
+
+    def test_meta_array_expander(self):
+        """Test `meta_array_expander` no array elements."""
+
+        self.out = io.meta_array_expander(self.meta_dict)
+
+        # Ensure items unchanged
+        assert np.all(self.out == self.meta_dict)
+
+        return
+
+    def test_meta_array_expander_with_array_elements(self):
+        """Test `meta_array_expander` with array elements."""
+
+        # Add array elements
+        for var in self.meta_dict.keys():
+            self.meta_dict[var]['array_test'] = [1, 2, 3, 4]
+
+        # Apply test function
+        self.out = io.meta_array_expander(self.meta_dict)
+
+        # Confirm there is a change.
+        assert np.all(self.out != self.meta_dict), 'Return dict same as input.'
+
+        # Confirm array expansion
+        estr = 'Missing expansion of array labels'
+        estr2 = 'Missing values in expanded labels'
+        for var in self.out.keys():
+            for tvar in np.arange(4):
+                tstr = 'array_test{:1d}'.format(tvar)
+                assert tstr in self.out[var], estr
+                assert self.out[var][tstr] == tvar + 1, estr2
```

Comment:
Error strings should provide more information about the variables being compared.